hw/xbox/nv2a/nv2a_pgraph.c

```diff
@@ -3041,6 +3059,40 @@ static void pgraph_bind_shaders(PGRAPHState *pg)
         }
     }
 
+    /* Window clip */
```

Comment:
nit: should mention that this first part of the algorithm is mostly an optimization:

- Don't do anything if the entire window  [`all_x`, `all_y`] is clipped/unclipped
- Remove duplicated clipping regions to avoid useless logic in fragment shader
    - D3D clones the last active clip region into all following slots (if I remember correctly)

The code should still work without this (just `state.psh.window_clip_count = 8;`).
Wether I was going to keep this optimization was still left to decide I believe (due to added code complexity / likely little performance gain).

Reply:
I think comment about this being an optimization is already pretty clear. Here's what the code says:

    /* Check if the last region is duplicated and ignore duplicates. We'll also
     * early out on regions which cover the entire surface.
     */

To be this is pretty clear, but I can emphasize that this is an optimization by us to detect any blatantly overlapping regions.

Reply:
Yeah, that'd be good. I think it's not clear that this code would still work fine if that block was removed.

Maybe also mention that we are not sure if it helps with performance?
If future-us will refactor this we might want to get rid of it to reduce complexity at 0-cost.

Wether you do this change is up to you - it's a minor doc issue.